tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/ModelValueHandler.scala

```diff
@@ -9,37 +9,49 @@ import scala.util.matching.Regex
  * types (and SMT sorts) are unique and they are incomparable with regular strings of with other
  * model values of a different type.
  *
- * Each model value follows the pattern
- *  [PREFIX]_[TYPE]_[INDEX]
- * where:
- *  - PREFIX is a prefix shared by, and identifying, all model values.
- *  - TYPE is the type of this particular model value. e.g., a value [PREFIX]_XYZ_[INDEX] has the uninterpreted type XYZ
- *  - INDEX is a unique identifier within the model value type. Its actual value is unimportant, however,
- *  model values with different indices are unequal by definition
+ * Each model value follows the pattern: `[NAME]_OF_[TYPE]`, where:
```

Comment:
You know, since we're dealing with strings, why not make the syntax literally `[NAME] : [TYPE]`? `:` is a character too.

Reply:
I like your out-of-the-box thinking :))